modules/apps/27-interchain-accounts/controller/ibc_module.go

```diff
@@ -42,16 +42,16 @@ func (im IBCModule) OnChanOpenInit(
 	chanCap *capabilitytypes.Capability,
 	counterparty channeltypes.Counterparty,
 	version string,
-) error {
+) (string, error) {
 	if !im.keeper.IsControllerEnabled(ctx) {
-		return types.ErrControllerSubModuleDisabled
+		return "", types.ErrControllerSubModuleDisabled
 	}
 
 	if err := im.keeper.OnChanOpenInit(ctx, order, connectionHops, portID, channelID, chanCap, counterparty, version); err != nil {
-		return err
+		return "", err
 	}
 
-	// call underlying app's OnChanOpenInit callback with the appVersion
+	// call underlying app's OnChanOpenInit callback with the passed in version
 	return im.app.OnChanOpenInit(ctx, order, connectionHops, portID, channelID,
 		chanCap, counterparty, version)
```

Comment:
This gives an opportunity to the interchain accounts auth module to modify the version string, or perhaps mistakenly return the incorrect string, right? 

Is this something we want to support? i.e. providing the auth module the ability to mutate this? 
Currently the only expectation from this callback is the claiming of the channel capability.

The code below would complain about `appVersion` declared but not used...

```suggestion
        // call underlying app's OnChanOpenInit callback with the passed in version
        appVersion, err := im.app.OnChanOpenInit(ctx, order, connectionHops, portID, channelID,
		chanCap, counterparty, version)
	if err != nil {
	    return "", err
	}
	
	return version, nil
```

Reply:
Not really following your point with the code example. 

What would be the alternative here?

Reply:
Agreed if ibc-auth is expected not to change the version at all. Then we should ignore it's return value. And if relayer passes in empty string, then just return our default version

Reply:
Sorry if that was confusing, like @AdityaSripal says, we should ignore the `appVersion` field from the code example above by discarding it, as we don't want to depend on ica-auth to return the correct string(`version`) or risk modifying it:

```
_, err := im.app.OnChanOpenInit(ctx, order, connectionHops, portID, channelID, chanCap, counterparty version)
if err != nil {
    return "", err
}
```

> And if relayer passes in empty string, then just return our default version

But for ICA a relayer is never expected to initiate the channel handshake, it should be from some on-chain mechanism, correct?

Reply:
Updated this. Good catch :+1: